### ISSUE TITLE
archive.extracted tar_options can't handle long options properly

### ISSUE THREAD

**Issue by cjbarnes18** [saltstack/salt#19799]
I managed to get it to work bit it's quite obscure.

```yaml
node:
  archive.extracted:
    - name: /usr/local/
    - source: http://nodejs.org/dist/v0.10.35/node-v0.10.35-linux-x64.tar.gz
    - source_hash: sha1=3a202a749492e48542d2c28220e43ef6dae084bc
    - archive_format: tar
    - tar_options: z --strip-components=1 -
    - if_missing: /usr/local/bin/node
```
long options should be possible without this obscurity.

**Comment by jfindlay** [comment]
@cjbarnes18, thanks for reporting this.

**Comment by MisterRager** [comment]
I was able to use `-strip-components=1` successfully.

**Comment by MisterRager** [comment]
I take that back, when I do that, the error is now with the current working directory not existing:

```
----------
          ID: /usr/share/solr
    Function: archive.extracted
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1542, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/states/archive.py", line 210, in extracted
                  results = __salt__['cmd.run_all'](tar_cmd, cwd=name, python_shell=False)
                File "/usr/lib/python2.7/dist-packages/salt/modules/cmdmod.py", line 892, in run_all
                  use_vt=use_vt)
                File "/usr/lib/python2.7/dist-packages/salt/modules/cmdmod.py", line 360, in _run
                  .format(cwd)
              CommandExecutionError: Specified cwd '/usr/share/solr' either not absolute or does not exist
     Started: 00:31:56.130617
    Duration: 3.325 ms
     Changes:   
```

If the directory does not exist, it is not extracted due to this error. If it does exist, it is not extracted because salt will not replace the existing directory.

**Comment by jfindlay** [comment]
@cjbarnes18, the archive state handling of tar options has been updated since 2014.7.0 and should be able to work with all kinds of interesting tar options.  Can you confirm that this is fixed?  Thanks.

**Comment by rallytime** [comment]
@cjbarnes18 Have you been able to give the tar options a try on a newer release? We'd like to get this one closed up if possible, or fixed if something isn't quite right. Thanks!

**Comment by rallytime** [comment]
Since we never heard back here and I am pretty sure this is fixed in latest stable (2015.5.2) I am going to close this. If this remains a problem, let us know and we will gladly open this issue again. Thanks!
